utils/post_processing/bfiq_to_rawacf.py

```diff
@@ -0,0 +1,157 @@
+		dims = ts_dict["data_dimensions"]
+
+		lags = ts_dict["lags"]
+		num_lags = np.shape(ts_dict["lags"])[0]
+		num_ranges = ts_dict["num_ranges"]
+		num_slices = ts_dict["num_slices"]
+		
+		data_mat = data_buff.reshape(dims)
+
+		# Get data from each antenna array
+		main_data = data_mat[0][:][:][:]
+		intf_data = data_mat[1][:][:][:]
+
+		# Preallocate arrays for correlation results
+		main_corrs = np.zeros((num_sequences, num_beams, num_samples, num_samples), dtype=np.complex64)
+		intf_corrs = np.zeros((num_sequences, num_beams, num_samples, num_samples), dtype=np.complex64)
+		cross_corrs = np.zeros((num_sequences, num_beams, num_samples, num_samples), dtype=np.complex64)
+
+		# Preallocate arrays for results of range-lag selection
+		out_main = np.zeros((num_sequences, num_beams, num_ranges, num_lags), dtype=np.complex64)
+		out_intf = np.zeros((num_sequences, num_beams, num_ranges, num_lags), dtype=np.complex64)
+		out_cross = np.zeros((num_sequences, num_beams, num_ranges, num_lags), dtype=np.complex64)
+
+		# Perform autocorrelations of each array, and cross
+		# correlation between arrays
+		for seq in range(num_sequences):
+			for beam in range(num_beams):
+				main_samps = main_data[seq, beam]
+				intf_samps = intf_data[seq, beam]
+
+				main_corrs[seq, beam] = np.outer(main_samps.conjugate(), main_samps)
+				intf_corrs[seq, beam] = np.outer(intf_samps.conjugate(), intf_samps)
+				cross_corrs[seq, beam] = np.outer(main_samps.conjugate(), intf_samps)
+
+				beam_offset = num_beams * num_ranges * num_lags
+				first_range_offset = int(ts_dict["first_range"] / ts_dict["range_sep"])
+
+				# Select out the lags for each range gate
+				main_small = np.zeros((num_ranges, num_lags,), dtype=np.complex64)
+				intf_small = np.zeros((num_ranges, num_lags,), dtype=np.complex64)
+				cross_small = np.zeros((num_ranges, num_lags,), dtype=np.complex64)
+
+				for rng in range(num_ranges):
+					for lag in range(num_lags):
+						# tau spacing in us, sample rate in hz
+						tau_in_samples = np.ceil(ts_dict["tau_spacing"] * 1e-6 * 
+													ts_dict["rx_sample_rate"])
+						p1_offset = lags[lag, 0] * tau_in_samples
+						p2_offset = lags[lag, 1] * tau_in_samples
+						
+						row_offset = int(rng + first_range_offset + p1_offset)
+						col_offset = int(rng + first_range_offset + p2_offset)
+
+						main_small[rng, lag] = main_corrs[seq, beam, row_offset, col_offset]
+						intf_small[rng, lag] = intf_corrs[seq, beam, row_offset, col_offset]
+						cross_small[rng, lag] = cross_corrs[seq, beam, row_offset, col_offset]
+
+				# replace full correlation matrix with resized range-lag matrix
+				out_main[seq, beam] = main_small
+				out_intf[seq, beam] = intf_small
+				out_cross[seq, beam] = cross_small
+
+		# average each correlation matrix over sequences dimension
+		out_main = np.mean(out_main, axis=0)
+		out_intf = np.mean(out_intf, axis=0)
+		out_cross = np.mean(out_cross, axis=0)
+
+		return out_main, out_intf, out_cross
```

Comment:
When you have a long method or loop or block of code and it's ending, it's a good practice to leave a comment saying what block is ending, here I might put:
`# def correlate_samples(ts_dict)`